src/shared/Microsoft.Git.CredentialManager/Authentication/MicrosoftAuthentication.cs

```diff
@@ -360,6 +398,30 @@ public HttpClient GetHttpClient()
 
         #region Auth flow capability detection
 
+        private bool CanUseBroker()
+        {
+            // We only support the broker on Windows 10 and .NET Framework
+#if NETFRAMEWORK
+            if (Context.SessionManager.IsDesktopSession && PlatformUtils.IsWindows10())
+            {
+                // Default to using the OS broker
+                const bool defaultValue = true;
+
+                if (Context.Settings.TryGetSetting(Constants.EnvironmentVariables.MsAuthUseBroker,
+                    Constants.GitConfiguration.Credential.SectionName,
+                    Constants.GitConfiguration.Credential.MsAuthUseBroker,
+                    out string valueStr))
+                {
+                    return valueStr.ToBooleanyOrDefault(defaultValue);
+                }
+
+                return defaultValue;
+            }
+#endif
+
+            return false;
```

Comment:
Personal preference: if there are ways to return from a function early, do so, as it reduces cognitive load for the reader. In this instance:

```c#
if (!!context.SessionManager.IsDesktopSession || !PlatformUtils.IsWindows10())
    return false;

[...]
```

Reply:
Updated.